osu.Game/Extensions/DrawableExtensions.cs

```diff
@@ -57,6 +58,9 @@ public static void ApplySkinnableInfo(this Drawable component, SkinnableInfo inf
             component.Anchor = info.Anchor;
             component.Origin = info.Origin;
 
+            if (component is ISkinnableDrawable skinnable)
+                skinnable.OverridesClosestAnchor = info.OverridesClosestAnchor;
```

Comment:
Not super keen on this type check, especially combined with the TODO above. Given that now `ISkinnableDrawable` is becoming more specific to the skinning setup than `Drawable`, I think a good way forward here would be to add the other values being applied in this method (ie. `Position`, `Rotation`, etc.) to `ISkinnableDrawable`. That way `SkinnableInfo` could also implement that interface, and there could be a general-purpose
```csharp
public static void CopyFrom(this ISkinnableInfo target, ISkinnableInfo source)
```
method that can be reused both here and in the `SkinnableInfo` ctor. Not super sure whether that should happen as part of this PR or not, though.

Reply:
The type check is definitely ugly. Do you want me to open a new PR that lessens the usage of `Drawable` in favor of `ISkinnableDrawable`?

Reply:
Not 100% sure myself. Would like input from others as to whether that sounds like a logical direction to take.

Reply:
I believe `SkinnableInfo` wouldn't even have to exist and the drawable would be serialised/populated using the `ISkinnableDrawable` interface? that I would agree to.

Reply:
Not entirely sure serialising interfaces is a thing that will work super well. May still need `SkinnableInfo` as a POCO intermediary if JSON libs will not comply.

Reply:
Yeah, [this](https://www.c-sharpcorner.com/UploadFile/20c06b/deserializing-interface-properties-with-json-net/) looks like a mess to me compared to just a POCO.